Linking labels

andrew-nowak opened the pull request:
Before merging, this PR requires
- [x] UX approval
- [x] Manual testing

Resolves #2753

~~Blocked on getting psammead utilities updated in simorgh~~
~~Blocked on translations for 'See All' on Pidgin, Igbo, Yoruba~~

**Overall change:** Pulls in linking section labels

**Code changes:**

- Add translations to all the services that have provided one (and use 'See All' for the rest)
- Update thai fixture data
- Conditionally wrap story promo list / single story promo with a margin-top when necessary
  - There is an almighty bodge to handle the fact that currently there are no desktop-width top stories. In #3198 (or this PR if that goes first), devs should unpick this bodge. (This is also why I haven't made more effort in avoiding it; anything I do needs unpicking in that PR so I made it straightforward and provided code to future devs/me :) ). 
- If there is a LINK in the strapline, render it

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [x] This PR requires manual testing

andrew-nowak committed:
add "see all" translations

andrew-nowak committed:
pull in new section label; update thai fixture data

andrew-nowak committed:
add stories

andrew-nowak committed:
Merge remote-tracking branch 'origin/latest' into linking-labels

andrew-nowak committed:
dont need defaults if all translations are done

andrew-nowak committed:
Merge remote-tracking branch 'origin/latest' into linking-labels

andrew-nowak committed:
snapshots

andrew-nowak committed:
Merge branch 'latest' into linking-labels

andrew-nowak committed:
fix the capitalisation

andrew-nowak committed:
add seeAll translations for first 3 fps

andrew-nowak committed:
Merge remote-tracking branch 'origin/latest' into linking-labels

andrew-nowak committed:
update snapshots

andrew-nowak committed:
Merge remote-tracking branch 'origin/latest' into linking-labels

# Conflicts:
#	package-lock.json
#	package.json

andrew-nowak committed:
dont emit react warnings and bodge desktop width

andrew-nowak committed:
Merge remote-tracking branch 'origin/latest' into linking-labels

andrew-nowak committed:
Merge branch 'latest' into linking-labels

DenisHdz reviewed (changes requested):
Can you add a story for Thai so we can also see linkeables straplines in Storybook?

andrew-nowak committed:
add a thai front page story

andrew-nowak committed:
Merge remote-tracking branch 'origin/latest' into linking-labels

andrew-nowak commented:
> Can you add a story for Thai so we can see a linkeable strapline also in Storybook?

There are 'link to more' stories in FrontPageSection - I assume you meant you'd also like Thai in the Front Page stories - makes sense now that page is live! added

andrew-nowak committed:
Merge remote-tracking branch 'origin/latest' into linking-labels

# Conflicts:
#	data/thai/frontpage/index.json
#	src/app/lib/utilities/preprocessor/rules/__snapshots__/topstories.test.js.snap

DenisHdz commented on src/app/containers/FrontPageSection/index.jsx:
```diff
@@ -9,6 +12,40 @@ import groupShape from '../../models/propTypes/frontPageGroup';
+  //  At that point, delete the below return statement and uncomment the following.
+
+  // return children;
+  return <TopMarginSixHundred>{children}</TopMarginSixHundred>;
```
Do we need the if? Can't we have just one return with the `as` attribute which it will be set it we pass it via props?
```
return <TopMarginSixHundred as={as}>{children}</TopMarginSixHundred>;
```

andrew-nowak replied:
In the current bodged state, you can get away with it, yeah. But you need to follow the commented out code for how this component should behave if desktop width top stories already existed, which needs that check. And I'm trying to keep the code as close as possible to the desired result

DenisHdz commented on src/app/containers/FrontPageSection/index.jsx:
```diff
@@ -57,11 +97,13 @@ const FrontPageSection = ({ bar, group, sectionNumber }) => {
       </SectionLabel>
       {items.length > 1 ? (
-        <StoryPromoUl>
+        <MarginWrapper addMargin={!isFirstSection} as={StoryPromoUl}>
```
Since both `TopMargin` and `TopMarginSixHundred`, add margin to the top, what about calling the prop `firstSection` instead of `addMargin`? And then in `MarginWrapper` do `if (firstSection) { ... }`

andrew-nowak replied:
imo this way the props are more descriptive - the MarginWrapper just wraps a component with a `margin-top` - it has no idea what a section or story promo or story promo list is, it just wraps the children its passed. The only logic of whether or not to add the margin is contained in setting the prop `addMargin={!isFirstSection}`, ie. the logic is owned by the FrontPageSection

DenisHdz replied:
Ok, you convinced me :)

DenisHdz committed:
Merge branch 'latest' into linking-labels

DenisHdz reviewed (approved):
![](https://media.giphy.com/media/5tmRQlzMmsLxMvKln8/giphy.gif)

EinsteinNjoroge committed:
Merge branch 'latest' into linking-labels

sadickisaac commented on src/app/containers/FrontPageSection/index.jsx:
```diff
@@ -9,6 +12,40 @@ import groupShape from '../../models/propTypes/frontPageGroup';
+const MarginWrapper = ({ addMargin, as, children }) => {
+  // Conditionally add a `margin-top` to the `children`, mixing in
+  // `as` if present.
+  if (addMargin) {
```
```jsx
addMargin && <TopMargin /*....*/ />
as && <TopMarginSixHundred /* ... */ />
return <TopMarginSixHundred /*... */ />
```
Any thoughts on this

Commit:
Merge branch 'latest' into linking-labels

EinsteinNjoroge committed:
Merge branch 'latest' into linking-labels

andrew-nowak committed:
Merge remote-tracking branch 'origin/latest' into linking-labels

# Conflicts:
#	data/thai/frontpage/index.json
#	src/app/lib/config/services/afaanoromoo.js
#	src/app/lib/config/services/afrique.js
#	src/app/lib/config/services/amharic.js
#	src/app/lib/config/services/arabic.js
#	src/app/lib/config/services/azeri.js
#	src/app/lib/config/services/bengali.js
#	src/app/lib/config/services/burmese.js
#	src/app/lib/config/services/cymrufyw.js
#	src/app/lib/config/services/gahuza.js
#	src/app/lib/config/services/gujarati.js
#	src/app/lib/config/services/hausa.js
#	src/app/lib/config/services/hindi.js
#	src/app/lib/config/services/indonesia.js
#	src/app/lib/config/services/japanese.js
#	src/app/lib/config/services/korean.js
#	src/app/lib/config/services/kyrgyz.js
#	src/app/lib/config/services/marathi.js
#	src/app/lib/config/services/mundo.js
#	src/app/lib/config/services/naidheachdan.js
#	src/app/lib/config/services/nepali.js
#	src/app/lib/config/services/pashto.js
#	src/app/lib/config/services/persian.js
#	src/app/lib/config/services/portuguese.js
#	src/app/lib/config/services/russian.js
#	src/app/lib/config/services/serbian.js
#	src/app/lib/config/services/sinhala.js
#	src/app/lib/config/services/somali.js
#	src/app/lib/config/services/swahili.js
#	src/app/lib/config/services/tamil.js
#	src/app/lib/config/services/telugu.js
#	src/app/lib/config/services/tigrinya.js
#	src/app/lib/config/services/turkce.js
#	src/app/lib/config/services/ukchina.js
#	src/app/lib/config/services/ukrainian.js
#	src/app/lib/config/services/urdu.js
#	src/app/lib/config/services/uzbek.js
#	src/app/lib/config/services/vietnamese.js
#	src/app/lib/config/services/zhongwen.js
#	src/app/lib/utilities/preprocessor/rules/__snapshots__/topstories.test.js.snap

andrew-nowak committed:
Merge branch 'latest' into linking-labels

andrew-nowak committed:
updating bundle limits

andrew-nowak committed:
Merge branch 'latest' into linking-labels

andrew-nowak committed:
Merge branch 'latest' into linking-labels

PriyaKR committed:
Merge branch 'latest' into linking-labels

PriyaKR commented:
Testing:

See all link looks good(some of the services which dont have translations have the text in english).
Once the see all link is visited shouldn't it have a visited link text colour as compared to the unvisited link text colour?

Accessibility testing needs to be done.

aacn500 commented:
Thanks Priya! 
> See all link looks good(some of the services which dont have translations have the text in english).

This is known
> Once the see all link is visited shouldn't it have a visited link text colour as compared to the unvisited link text colour?
> 
> Accessibility testing needs to be done.

PriyaKR commented:
Ok thanks then accessibility testing needs to be done.

andrew-nowak commented:
Thanks Priya! 

> See all link looks good(some of the services which dont have translations have the text in english).

This is known, and is expected to be fixed as translations are provided before a given service goes live.

> Once the see all link is visited shouldn't it have a visited link text colour as compared to the unvisited link text colour?

Have double checked with UX, this is expected & desired by their design 
 
> Accessibility testing needs to be done.

An accessibility swarm was carried out on this component before release in Psammead, see https://github.com/bbc/psammead/pull/1534#issuecomment-521161190 for the details of that swarm :)

andrew-nowak committed:
Merge remote-tracking branch 'origin/latest' into linking-labels

# Conflicts:
#	src/app/containers/FrontPage/index.stories.jsx

andrew-nowak committed:
Merge branch 'latest' into linking-labels

PriyaKR commented:
Accessibility testing looks good to me.

PriyaKR committed:
Merge branch 'latest' into linking-labels

andrew-nowak committed:
Merge branch 'latest' into linking-labels

andrew-nowak committed:
Merge branch 'latest' into linking-labels

andrew-nowak commented:
it doesn't work on ie11
![image](https://user-images.githubusercontent.com/10963046/63588320-1a676880-c59e-11e9-9fa9-f7bfd1e55290.png)

back to drawing board

andrew-nowak committed:
Merge remote-tracking branch 'origin/latest' into linking-labels

# Conflicts:
#	scripts/bundleSize.sh

andrew-nowak committed:
bundle sizing

andrew-nowak committed:
Merge remote-tracking branch 'origin/latest' into linking-labels

andrew-nowak committed:
pull in ie11 fix

andrew-nowak commented:
Pulling back into 'Ready for Test' -- the only change since the last time in this column is pulling in this psammead fix for IE11, everything else should be identical :)

staylos92 committed:
Merge branch 'latest' into linking-labels

PriyaKR committed:
Merge branch 'latest' into linking-labels

andrew-nowak committed:
Merge remote-tracking branch 'origin/latest' into linking-labels

# Conflicts:
#	data/thai/frontpage/index.json
#	scripts/bundleSizeConfig.js
#	src/app/lib/utilities/preprocessor/rules/__snapshots__/topstories.test.js.snap

andrew-nowak committed:
Merge remote-tracking branch 'origin/latest' into linking-labels

# Conflicts:
#	scripts/bundleSizeConfig.js

andrew-nowak committed:
rerun snyk

PriyaKR committed:
Merge branch 'latest' into linking-labels

PriyaKR commented:
Looks good to me.

andrew-nowak committed:
Merge branch 'latest' into linking-labels
